### PR TITLE
added Machine Translation and Datasets(10.12)

### PR DESCRIPTION
**Pull request by gupta1912**
add 10.12

### CODE REVIEW

File: Ch10_Recurrent_Neural_Networks/Machine_Translation_and_Data_Sets.ipynb [hunk 1]
```diff
@@ -0,0 +1,1198 @@
...
+    "sys.path.insert(0, '..')\n",
+    "import collections\n",
+    "import d2l\n",
+    "import zipfile\n",
+    "from d2l.data.base import Vocab\n",
+    "\n",
+    "import torch\n",
+    "import torch.nn as nn\n",
+    "import torch.nn.functional as F\n",
+    "from torch.utils import data as gdata"
```
Comment (AnirudhDagar) [inline Ch10_Recurrent_Neural_Networks/Machine_Translation_and_Data_Sets.ipynb]: gdata is basically an import showing gluon data. We don't want this. Please keep this import as data. And similarly change the ther places.

File: Ch10_Recurrent_Neural_Networks/Machine_Translation_and_Data_Sets.ipynb [hunk 2]
```diff
@@ -0,0 +1,1198 @@
...
+    "colab_type": "code",
+    "id": "HfPtS21avd-t",
+    "outputId": "691fda1c-1f86-466e-bf49-22d31c095ae6"
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "--2019-06-08 15:18:12--  http://www.manythings.org/anki/fra-eng.zip\n",
```
Comment (AnirudhDagar) [inline Ch10_Recurrent_Neural_Networks/Machine_Translation_and_Data_Sets.ipynb]: Here make two different code blocks. 
One to `wget and reading the zip` data, and,
Other to print out `print(raw_text[0:95])`

**Comment by gupta1912** [comment]
Made required changes in 2nd commit

**Comment by AnirudhDagar** [comment]
Thanks! Great work. Will merge this. 😄

**Pull request closed without merging**